### PR TITLE
docs: create bot comparison page

### LINKED ISSUE #24121: Objective comparison between Dependabot and Renovate
### Describe the proposed change(s).

Have a page which shows the objective differences between Dependabot and Renovate. I'm thinking of things like:

- [x] Grouped updates
- [x] Dependency Dashboard
- [ ] Monorepo support
- [x] Supported platforms
- [x] Customizability
- [x] Show changelogs
- [x] Link to Dependabot documentation
- [x] Dependabot is built-in to GitHub, Renovate must be installed as an app (or self-hosted)
- [x] Compatiblity support badges
- [x] Update `.pages` with link to file so it's in the navigation (sidebar) on the published site

### Describe why we need/want these change(s).

A user recently asked for a comparison between features for Renovate and Dependabot:

- https://github.com/renovatebot/renovate/issues/21081#issuecomment-1685423278

`@rarkins` agreed that we could use a objective comparison.

### LINKED ISSUE #24689: Explain how Renovate and Dependabot show changelogs in bot comparison page
### Describe the proposed change(s).

Explain the differences between Renovate and Dependabot in how they show the changelogs. Relevant review comment:

> Besides change logs, commit history and release notes are included with Dependabot
>
> How does Renovate compare?

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Create bot comparison between Dependabot and Renovate
- Put new page in the `nav` configuration file

## Context

The goal is to show readers an _objective_ comparison between Dependabot and Renovate.

Closes #24121
Closes #24689 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/bot-comparison.md
Comment (HonkingGoose) [inline docs/usage/bot-comparison.md:11]: I don't know if Dependabot supports monorepos. The closest information I could find is [GitHub Docs, troubleshooting Dependabot errors](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/troubleshooting-dependabot-errors#dependabot-timed-out-during-its-update):

> ## Dependabot timed out during its update
>
> Dependabot took longer than the maximum time allowed to assess the update required and prepare a pull request. This error is usually seen only for large repositories with many manifest files, for example, npm or yarn monorepo projects with hundreds of package.json files. Updates to the Composer ecosystem also take longer to assess and may time out.

File: docs/usage/bot-comparison.md
Comment (HonkingGoose) [inline docs/usage/bot-comparison.md:34]: Same "Does Dependabot support monorepos" problem here. 😄

File: docs/usage/.pages
Comment (HonkingGoose) [inline docs/usage/.pages:3]: We'll need to decide _where_ to put this information in our docs sidebar.

I think it should go somewhere really visible (like the second page) or we'll still get questions like: "Do you guys have a comparison between Dependabot and Renovate"... 🙃

Comment (secustor) [reply docs/usage/.pages:3]: I would call the page "Renovate vs ...", that way we can add for example Snyk

File: docs/usage/bot-comparison.md
Comment (secustor) [inline docs/usage/bot-comparison.md:17]: We should add the most important ones and then link the platform overview

File: docs/usage/.pages
Comment (HonkingGoose) [reply docs/usage/.pages:3]: I agree we should have some generic name that can fit an expanded comparison. But I think we want to stay away from the vs word though to keep things fair? 🙂

Adding Snyk is something we can do later, let's focus on the Dependabot comparison in this PR.

File: docs/usage/bot-comparison.md
Comment (yeikel) [inline docs/usage/bot-comparison.md:16]: What does `Monorepo support` entail? Can this be elaborated in terms of what renovate supports?

File: docs/usage/bot-comparison.md
Comment (yeikel) [reply docs/usage/bot-comparison.md:23]: Also, we need to mention that dependabot also supports further customization using 

[schedule.day](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#scheduleday)

[schedule.timezone](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#scheduletimezone)

[schedule.time](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#scheduletime)

File: docs/usage/bot-comparison.md
Comment (yeikel) [reply docs/usage/bot-comparison.md:16]: Ah, reading further down, monorepo refers here to out of the box upgrade groups

File: docs/usage/bot-comparison.md
Comment (yeikel) [inline docs/usage/bot-comparison.md:1]: One more entry worth mentioning here is the license model. 

Renovate uses `GNU Affero General Public License.` 

Dependabot uses `Prosperity Public License 2.0.0`

File: docs/usage/bot-comparison.md
Comment (yeikel) [inline docs/usage/bot-comparison.md:20]: Besides change logs, commit history and release notes are included  with Dependabot.

File: docs/usage/bot-comparison.md
Comment (yeikel) [inline docs/usage/bot-comparison.md:83]: This should be expanded further for self-hosted teams. 

For Dependabot, the Github Enterprise license includes it for free

To add Renovate to a corporate environment there is a license cost associated as well as a separate infra needed

Comment (yeikel) [reply docs/usage/bot-comparison.md:83]: Additionally, the "Dependabot alerts" included in Github are highly connected to Dependabot bot

Comment (HonkingGoose) [reply docs/usage/bot-comparison.md:83]: > Additionally, the "Dependabot alerts" included in Github are highly connected to Dependabot bot

You can select what parts of Dependabot you want to use in the repository settings. Renovate uses "Dependabot alerts" and it's "Dependency Graph" for Renovate's [`vulnerabilityAlerts`](https://docs.renovatebot.com/configuration-options/#vulnerabilityalerts) feature, but the Renovate users _disables_ the normal Dependabot upgrades.

Comment (HonkingGoose) [reply docs/usage/bot-comparison.md:83]: > This should be expanded further for self-hosted executions in corporate environments.
> 
> For Dependabot, the Github Enterprise license includes it for free

Enterprises can use Renovate for free as well on GitHub. Quote from the Renovate GitHubapp installation page: [^1]

> ### Mend Free Plan
>
> Enjoy Renovate as a service for free, hosted by mend.io
>
> ✅  Private and Public repositories 

---

> To add Renovate to a corporate Github environment there is a license cost associated as well as a separate infra

If you go the self-hosting route, you'll have to pay for the server, and deal with upgrading the open source Renovate packages/container yourself. 😄 That would be no different if you decided to self-host a GitHub instance.

I think there's a possible confusion here for readers between the _paid_ Enterprise offerings of Mend, and the free opensource Renovate.

[^1]: https://github.com/marketplace/renovate

File: docs/usage/bot-comparison.md
Comment (HonkingGoose) [reply docs/usage/bot-comparison.md]: Thank you for the explanation, I reworded the table and the sentence about monorepos.

File: docs/usage/bot-comparison.md
Comment (andrcuns) [inline docs/usage/bot-comparison.md:49]: This is only true if compared with native `GitHub` integration.

The open-source component [dependabot-core](https://github.com/dependabot/dependabot-core) does allow creating solutions for other platforms.

* currently broken but I believe it used to support gitlab, bitbucket and probably other platforms: https://github.com/dependabot/dependabot-script
* community maintained (mostly by me) version that works with GitLab: https://gitlab.com/dependabot-gitlab/dependabot

Disclaimer, I'm not a `dependabot` maintainer or otherwise associated with `dependabot` team or `GitHub`

File: docs/usage/bot-comparison.md
Comment (rarkins) [reply docs/usage/bot-comparison.md:87]: I'm mostly staying out of this but will participate to correct factual errors like below.

> To add Renovate to a corporate Github environment there is a license cost associated

Incorrect

> as well as a separate infra

Incorrect: you can run Renovate using GitHub Actions which means no separate infra

> Additionally, the "Dependabot alerts" included in Github are highly connected to Dependabot bot

This too vague to be included in any comparison. Stick to specifics

File: docs/usage/bot-comparison.md
Comment (rarkins) [reply docs/usage/bot-comparison.md:49]: Any comparison should indeed mention these, although be as impartially correct as possible about the others not being _supported_ in any official way. The first repo being broken for 6+ months kind of speaks for itself

Comment (andrcuns) [reply docs/usage/bot-comparison.md:49]: For sure, can't disagree here, but still, `Dependabot only works on GitHub.` is somewhat incorrect, maybe addition that `officially supported part only works on GitHub` or somewhere along those lines.

I imagine it's in the similar realm as `Renovate comes with community-provided groupings of dependencies, so it will group common dependencies out-of-the-box.` though I do imagine that `renovate` team is probably a bit more involved in supporting those community provided groupings 😄

File: docs/usage/bot-comparison.md
Comment (yeikel) [reply docs/usage/bot-comparison.md:87]: I stand corrected. I was under the impression that to run Renovate in a corporate environment, a license was needed

A few point stand however as running using Github Actions still involves having GitHub Actions runners (therefore, dedicated infra)

Comment (yeikel) [reply docs/usage/bot-comparison.md:87]: > That would be no different if you decided to self-host a GitHub instance.

That's correct. But there is no separate _maintenance needed explicitly for dependabot_

File: docs/usage/bot-comparison.md
Comment (yeikel) [reply docs/usage/bot-comparison.md:49]: I think that the distinction here should highlight that the Dependabot BOT (not core) only works in Github. The average user won't be able to setup dependabot-core and create a bot using it. It is possible(I have done it) but it takes a considerable effort to do so. 

In that direction, one other point is that renovate is fully open source while the full dependabot BOT is not. Only the core framework is open and available to be used for further customizations  as @andrcuns mentioned,

File: docs/usage/bot-comparison.md
Comment (yeikel) [inline docs/usage/bot-comparison.md:2]: I am not sure if it is worth mentioning it, but Dependabot is written in Ruby while Renovate is in Javascript 

For users looking to be future contributors or to extend the bots that might matter

File: docs/usage/bot-comparison.md
Comment (yeikel) [reply docs/usage/bot-comparison.md]: I think that the explanation is a bit misleading still. 

Rather than `Upgrades common monorepo packages at once` 

I'd reword it as "Out of the box configuration for common monorepo packages" 

Dependabot can support this using the groups configuration so stating that it is not supported is misleading

File: docs/usage/bot-comparison.md
Comment (yeikel) [inline docs/usage/bot-comparison.md:21]: I think that this should be expanded a bit further as the number does not mean much for someone new to the products

File: docs/usage/bot-comparison.md
Comment (yeikel) [inline docs/usage/bot-comparison.md:22]: `No, requires app or self-hosting`

We can add a link OR explain what self-hosting means

Also, it is probably worth mentioning that there is paid offering for this

File: docs/usage/bot-comparison.md
Comment (yeikel) [inline docs/usage/bot-comparison.md:19]: Maybe it'll be too much to ask, but it'd help to break this down further. 

What can we customize in Renovate that we can't in Dependabot?

File: docs/usage/bot-comparison.md
Comment (rarkins) [reply docs/usage/bot-comparison.md:87]: > Under what situations a license is needed if at all? Can this be expanded?

Renovate is Open Source. The license is right here in the project. There's not much expansion necessary beyond "Renovate is Open Source licensed"

> A few point stand however as running using Github Actions still involves having GitHub Actions runners (therefore, dedicated infra). Although it is fair to say that Github Action runners would still be needed for other workflows anyways

This is also an invalid/inaccurate point.

image

Comment (rarkins) [reply docs/usage/bot-comparison.md:87]: > > That would be no different if you decided to self-host a GitHub instance.
> 
> That's correct. But there is no separate _maintenance needed explicitly for dependabot_. Github Enterprise upgrades include dependabot bundled

It's not correct to say there's no separate maintenance. There's multiple pages on managing Dependabot for GHES in addition to any regular setup for GHES + Actions:
- https://docs.github.com/en/enterprise-server@3.7/admin/configuration/configuring-github-connect/enabling-dependabot-for-your-enterprise
- https://docs.github.com/en/enterprise-server@3.7/admin/github-actions/enabling-github-actions-for-github-enterprise-server/managing-self-hosted-runners-for-dependabot-updates

I actually think it would be faster to get a Renovate bot GitHub Action running on GitHub Enterprise Server than to follow those steps describe in the above links, but I'm not proposing to add that argument to comparison we have here. Renovate also doesn't require GitHub Actions to work with GHES either, although I'm not sure what % of enterprise self-hosters don't have Actions by now, so it may be a moot point.

File: docs/usage/bot-comparison.md
Comment (HonkingGoose) [reply docs/usage/bot-comparison.md]: Improved the table so it has a short summary of the different badges. 😉

File: docs/usage/bot-comparison.md
Comment (yeikel) [reply docs/usage/bot-comparison.md:87]: With "separate maintenance" I meant that there is no separate maintenance needed to the average user as that will be managed by the instance admins

But fair enough, it is not worth getting into that

Comment (yeikel) [reply docs/usage/bot-comparison.md:87]: I finally got some clarity to explain where my confusion came from. The reason why many organizations consider Renovate to be a paid product before it can be added safely to an organization is due to its licensing  which is highlighted(intentionally or unintentionally)  when we see that the Mend Renovate On-Premises [uses a difference license model](https://docs.renovatebot.com/getting-started/running/#mend-renovate-on-premises). 

Google and many other organizations ban explicitly the usage of [AGPL](https://opensource.google/documentation/reference/using/agpl-policy)

I am not a lawyer or qualified to discuss if the reasons are valid or not, but that's where my misunderstanding came from. I am sorry for the confusion and the misleading statements corrected above. 

Therefore, Renovate is indeed open source and usable in a corporate environment, but the license might not be allowed everywhere. Any developer planning to use Renovate behind a corporate environment needs to crosscheck that the license is allowed on their organizations and mentioning the license on this comparison (As well as other places) is more than enough

Thanks!

Comment (rarkins) [reply docs/usage/bot-comparison.md:87]: @yeikel you seem still wildly confused

> The reason why many organizations consider Renovate to be a paid product

You're the only person I can remember who's tried to claim that Renovate isn't free, and I don't think you have any reason to claim to speak for "many organizations".

You then justify that by mentioning Renovate On-Prem and its "license model", even though Renovate On-Prem has been available _for free_ under a very standard Enterprise-friendly EULA since 2019. 

And just to provide some grounding, your current employer is among the companies who have in the past requested one of the _free_ Renovate On-Prem licenses. Meanwhile Google, who you mention in a negative light, authorized themselves to be listed as a Renovate App reference customer on https://github.com/marketplace/renovate

Comment (yeikel) [reply docs/usage/bot-comparison.md:87]: I am not a lawyer and perhaps I should have said that AGPL is not allowed _in some organizations_ with Google being an example. The Renovate source code discussed in this repo is AGPL and I do not believe that saying that is misleading

Other than that, I was unaware of the process to get a free key for the self-hosted version and what rights and limitations (if any) it gives. The [request form](https://www.mend.io/free-developer-tools/renovate/on-premises/) says that it is free but asks for a lot of personal information and to me that gives the impression that it might be the start of a sale. If that's not the case, I stand corrected as well

Lastly, and to clarify, I do not speak in name of my employer nor I know what business they have with you. My opinions and understanding (even if misled) are my own

Thanks for clarifying!

File: docs/usage/bot-comparison.md
Comment (yeikel) [inline docs/usage/bot-comparison.md:21]: Would it be possible to have hyperlinks to each of these badges explaining what they mean?

File: docs/usage/bot-comparison.md
Comment (carogalvin) [inline docs/usage/bot-comparison.md:71]: Might be worthwhile to spell out some specifics here. This kind of gives the impression that there are _no_ customizable options for this with Dependabot.

**Comment by carogalvin** [comment]
Hey, I'm the product manager for Dependabot over at GitHub. I think this comparison overall is pretty fair. Now, I don't know how you'd feel about including this in this doc, but one of Dependabot's big selling points is our security-related updates and native integration with alerts and our advisory database, could be worthwhile spelling out in the comparison table? 🤷‍♀️

**Comment by HonkingGoose** [comment]
> Hey, I'm the product manager for Dependabot over at GitHub. I think this comparison overall is pretty fair.

Thank you very much for taking the time to review my work! ❤️ I'm glad that the comparison is pretty fair.

>  Now, I don't know how you'd feel about including this in this doc, but one of Dependabot's big selling points is our security-related updates and native integration with alerts and our advisory database, could be worthwhile spelling out in the comparison table? 🤷‍♀️

I completely forgot to add a security comparison to the table! Having the "security stuff" integrated into GitHub is very nice, so we should mention that.

The tricky part is that Renovate _extends_ at least one feature from Dependabot: Renovate can read GitHub's vulnerability alerts to customize Renovate's PRs. For that to work, the user must enable the Dependency Graph and Dependabot alerts. They should _not_ run Dependabot, or they'll get two updates for the same thing... 🙃 See the [`vulnerabilityAlerts` docs for Renovate](https://docs.renovatebot.com/configuration-options/#vulnerabilityalerts) if you want to know more.

Renovate also has a experimental feature called [`osvVulnerabilityAlerts`](https://docs.renovatebot.com/configuration-options/#osvvulnerabilityalerts) to check for vulnerabilities on the OSV database (open source vulnerability database).

**Comment by carogalvin** [comment]
> The tricky part is that Renovate extends at least one feature from Dependabot: Renovate can read GitHub's vulnerability alerts to customize Renovate's PRs. For that to work, the user must enable the Dependency Graph and Dependabot alerts.

🤔 so something like a category of "automated PRs to resolve security alerts" where Dependabot is like ✅ and Renovate is like " ✅ for GitHub when Dependabot alerts & dependency graph are enabled, OR when using experimental feature `osvVulnerabilityAlerts` "? Not sure if you have specific logic or configuration for security updates vs normal version bumps but could call out anything here as well.

File: docs/usage/bot-comparison.md
Comment (rarkins) [reply docs/usage/bot-comparison.md:71]: Agreed, let's avoid generalizations and use (meaningful) specifics instead - or leave such comparisons out

File: docs/usage/bot-comparison.md
Comment (HonkingGoose) [inline docs/usage/bot-comparison.md:1]: Reminder for myself to add a security section.

File: docs/usage/bot-comparison.md
Comment (HonkingGoose) [reply docs/usage/bot-comparison.md:73]: I'll leave it out, I don't know off-hand how to make a meaningful comparison, and I don't want to hold up this PR too long.

File: docs/usage/bot-comparison.md
Comment (HonkingGoose) [reply docs/usage/bot-comparison.md:58]:
```suggestion
The _official_ Dependabot program only works on GitHub.

If you're an advanced user, you may use the [`dependabot-core` repository](https://github.com/dependabot/dependabot-core) as a base to build your own Dependabot, which you can run on other platforms.
```

File: docs/usage/bot-comparison.md
Comment (HonkingGoose) [reply docs/usage/bot-comparison.md]: I first need hands-on experience with Dependabot to make a meaningful comparison. See issue:

- https://github.com/renovatebot/renovate/issues/24689

**Comment by HonkingGoose** [comment]
@rarkins I'm ready for a final round of review, and then a merge. I'd like to get at least something out the door. 🙃 

I'll try to do the issues below, after I get some hands-on experience with Dependabot:

- #24688
- #24689

File: docs/usage/bot-comparison.md
Comment (rarkins) [inline docs/usage/bot-comparison.md:158]:
```suggestion
The Dependabot team has a community-maintained collection of scripts to start self-hosting Dependabot: [`dependabot-script`](https://github.com/dependabot/dependabot-script) but the repository has included a message warning that the scripts are broken since March 2023.
```

File: docs/usage/bot-comparison.md
Comment (rarkins) [inline docs/usage/bot-comparison.md:118]:
```suggestion
Dependabot uses [The Prosperity Public License 2.0.0](https://github.com/dependabot/dependabot-core/blob/main/LICENSE).

Neither license is relevant to the end user though if consuming through an App/SaaS.

```

File: docs/usage/bot-comparison.md
Comment (rarkins) [inline docs/usage/bot-comparison.md:147]:
```suggestion
You can self-host Dependabot on other platforms than GitHub but none are officially supported.
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**
